File: Crio/Array/11_SearchA2DMatrix/SearchA2DMatrix.cpp
#include <bits/stdc++.h>
using namespace std;
#include "Solution.cpp"

// CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
// CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);
    int r, c;
    cin >> r >> c;
    vector<vector<long long>  > matrix(r, vector<long long >(c, 0));
    for (int i = 0; i < r; i++)
    {
        for (int j = 0; j < c; j++)
            cin >> matrix[i][j];
    }
    int q;
    cin >> q;
    while (q--)
    {
        long long target;
        cin >> target;
        bool result  =  answer(matrix, target);
        if (result)
            cout << "true" << "\n";
        else
            cout << "false\n";
    }
    return 0;
}

File: Crio/Array/11_SearchA2DMatrix/SearchA2DMatrix.py
import sys
from Solution import solve

# CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
# CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

def main():
    rows, columns = map(int, sys.stdin.readline().split())
    matrix = list()
    for _ in range(rows):
        row = list(map(int, sys.stdin.readline().split()))
        matrix.append(row)
    queries = int(sys.stdin.readline())
    for _ in range(queries):
        target = int(sys.stdin.readline())
        result = solve(matrix, target)
        if result:
            sys.stdout.write("true\n")
        else:
            sys.stdout.write("false\n")


if __name__ == "__main__":
    main()
